library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionAdapter.kt

```diff
@@ -60,7 +63,22 @@ internal class TransactionAdapter internal constructor(
                 path.text = "${transaction.method} ${transaction.getFormattedPath(encode = false)}"
                 host.text = transaction.host
                 timeStart.text = DateFormat.getTimeInstance().format(transaction.requestDate)
-                ssl.visibility = if (transaction.isSsl) View.VISIBLE else View.GONE
+
+                val sslDrawableRes = if (transaction.isSsl) {
+                    R.drawable.chucker_ic_https
+                } else {
+                    R.drawable.chucker_ic_http
+                }
+                val sslDrawableColorRes = if (transaction.isSsl) {
+                    R.color.chucker_color_primary
+                } else {
+                    R.color.chucker_color_error
+                }
+                ssl.setImageDrawable(AppCompatResources.getDrawable(itemView.context, sslDrawableRes))
+                ImageViewCompat.setImageTintList(
+                    ssl,
+                    ColorStateList.valueOf(ContextCompat.getColor(itemView.context, sslDrawableColorRes))
+                )
```

Comment:
nit: this could maybe be refactored to a sealed class to store the protocol logic (rather than having two similar if-then-else), something like?

```kotlin
sealed class Protocol(val icon : Int, val color: Int) {
        class Http() : Protocol(R.drawable.chucker_ic_http, R.color.chucker_color_primary)
        class Https() : Protocol(R.drawable.chucker_ic_https, R.color.chucker_color_primary)
        
        fun populateImage(imageDrawable: Drawable) {
            ...
        }
    }
```

Reply:
I did it in slightly other way, so the sealed class has only resources without function to populate image to not mix responsibilities. Image populating is still up to adapter class.

Reply:
I did it in slightly other way, so the sealed class has only resources without function to populate image to not mix responsibilities. Image populating is still up to adapter class.